helpers/hyper-v-mutating-webhook/webhook.go

```diff
@@ -73,11 +77,15 @@ func (a *podUpdater) Handle(ctx context.Context, req admission.Request) admissio
 	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
 }
 
-// podUpdater implements admission.DecoderInjector.
-// A decoder will be automatically injected.
-
-// InjectDecoder injects the decoder.
-func (a *podUpdater) InjectDecoder(d *admission.Decoder) error {
-	a.decoder = d
+// InjectDecoder injects a decoder into the podUpdater
+func (pu *podUpdater) InjectDecoder(d *admission.Decoder) error {
+	pu.decoder = d
 	return nil
 }
+
+func isHostProcessPod(p *corev1.Pod) bool {
+	if p.Spec.SecurityContext != nil && p.Spec.SecurityContext.WindowsOptions != nil && p.Spec.SecurityContext.WindowsOptions.HostProcess != nil {
```

Comment:
do we need to check the containers too or by the time the webhook is called the API server has already normalized this?

Reply:
All of the e2e tests set `hostProcess` true at the pod level.
For completeness I can update this logic to check if any of the containers also set it.

Reply:
updated